client/object_delete.go

```diff
@@ -61,6 +63,21 @@ func (x PrmObjectDelete) GetSession() (*session.Object, error) {
 	return &sess, nil
 }
 
+// IsAutoSessionEnabled is an indicator for Pool.
```

Comment:
But it's a `client` thing, it shouldn't care about `pool`.

We already have `WithinSession`, it can override `pool` behavior for the case when the session is used. We only need a way to explicitly _remove_ the same session. Either it's a `WithinSession(nil)` or a separate method (`DropSession()`). Then it can be a `bool` internally (`isSessionSet`) and maybe a special `error` from `GetSession()` to separate `ErrNoSession` and `ErrNoSessionExplicitly`. Then all of this is completely generic `client` thing, but `pool` can use these functions as needed for its magic.

Reply:
Renamed _DropSession_ with _IgnoreSession_, I think this is more neutral. But I tend to think we need to add some description, that this function does nothing for a single client

Reply:
It can do something for a single client! Suppose you've used `WithinSession()` somewhere and now you can `IgnoreSession()` as well to remove it.

Naming is hard, we know, `IgnoreSession` is not bad, but ideally we should somehow signify that this function in many ways is an antonym to `WithinSession`.

Reply:
`WithinSession` / `IgnoreSession` looks like antipodes. But added some comments to func docs